### PR TITLE
community[minor]: Support configurable search parameters for Milvus

### PR DESCRIPTION
**Pull request by yokingma**
#### Description

- Support configurable search parameters for Milvus.
- Fixed some annotation errors.
- Refactored some code.

### CODE REVIEW

File: libs/langchain-community/src/vectorstores/milvus.ts [hunk 1]
```diff
@@ -108,33 +130,12 @@ export class Milvus extends VectorStore {
...
-  indexSearchParams = JSON.stringify({ ef: 64 });
+  indexCreateParams: IndexCreateOptions;
 
-  _vectorstoreType(): string {
-    return "milvus";
-  }
+  indexSearchParams: keyValueObj;
 
-  constructor(embeddings: EmbeddingsInterface, args: MilvusLibArgs) {
+  constructor(public embeddings: EmbeddingsInterface, args: MilvusLibArgs) {
```
Comment (jacoblee93) [inline libs/langchain-community/src/vectorstores/milvus.ts:137]: Why change this?

File: libs/langchain-community/src/vectorstores/milvus.ts [hunk 2]
```diff
@@ -73,6 +75,22 @@ const MILVUS_VECTOR_FIELD_NAME = "langchain_vector";
 const MILVUS_TEXT_FIELD_NAME = "langchain_text";
 const MILVUS_COLLECTION_NAME_PREFIX = "langchain_col";
 
+/**
+ * Default parameters for index searching.
+ */
+const DEFAULT_INDEX_SEARCH_PARAMS: Record<IndexType, IndexSearchParam> = {
```
Comment (jacoblee93) [inline libs/langchain-community/src/vectorstores/milvus.ts:81]: Where are these from? Can you add a docs link?

File: libs/langchain-community/src/vectorstores/milvus.ts [hunk 3]
```diff
@@ -335,9 +349,9 @@ export class Milvus extends VectorStore {
       collection_name: this.collectionName,
       search_params: {
         anns_field: this.vectorField,
-        topk: k.toString(),
+        topk: k,
```
Comment (jacoblee93) [inline libs/langchain-community/src/vectorstores/milvus.ts:352]: This isn't necessary anymore?

**Comment by jacoblee93** [comment]
A few changes that look breaking unless Milvus updated their APIs - can try it myself if you add a test?

File: libs/langchain-community/src/vectorstores/milvus.ts (hunk 3 above)
Comment (zhanshuyou) [reply libs/langchain-community/src/vectorstores/milvus.ts:352]: topk is a number
image

**Review by zhanshuyou — APPROVED** [review]

File: libs/langchain-community/src/vectorstores/milvus.ts (hunk 3 above)
Comment (zhanshuyou) [reply libs/langchain-community/src/vectorstores/milvus.ts:352]: topk is a number
image

**Comment by jacoblee93** [comment]
Thank you @zhanshuyou!

**Pull request merged**